guiEditor/src/diagram/workbench.tsx

```diff
@@ -295,33 +295,46 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
         const draggedControl = this.props.globalState.draggedControl;
         const draggedControlParent = draggedControl?.parent;
         
-        if (draggedControlParent && draggedControl) {
-
-            draggedControlParent.removeControl(draggedControl);
-        
-            if (control != null) {
-                if (this.props.globalState.workbench.isContainer(control)) {
-                    this.props.globalState.guiTexture.removeControl(draggedControl);
-                    (control as Container).addControl(draggedControl);
+        //checking to make sure the element is not being inserted into a child
+        if (this._isNotChildInsert(control, draggedControl)) {
+            if (draggedControlParent && draggedControl) {
+                draggedControlParent.removeControl(draggedControl);
+
+                if (control != null) {

```

Comment:
would be great to comment on the different cases here as it might be hard to catch why we have so many ?